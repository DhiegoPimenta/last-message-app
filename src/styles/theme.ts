export default {
    grid: {
        container: '130rem',
        gutter: '3.2rem',
    },
    border: {
        radius: '1rem',
    },
    font: {
        family: `Poppins, Ubuntu, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif`,
        light: 300,
        normal: 400,
        bold: 600,
        extrabold: 800,
        sizes: {
            xsmall: '1.4rem',
            small: '1.8rem',
            medium: '2.4rem',
            large: '3.2rem',
        },
    },
    colors: {
        primary: '#64CBC5',
        secondary: '#0030FF',
        terciary: '#A0DFDC',
        white: '#FFFFFF',
        gray: '#CECECE',
        danger: '#FBA9B0',
        black: '#444444',
        overlay: 'rgba(0, 0, 0, 0.3)',
    },
    spacings: {
        xxsmall: '0.8rem',
        xsmall: '1.6rem',
        small: '2.4rem',
        medium: '3.2rem',
        large: '4.0rem',
        xlarge: '4.8rem',
        xxlarge: '5.6rem',
    },
    layers: {
        base: 10,
        menu: 20,
        overlay: 30,
        modal: 40,
        alwaysOnTop: 50,
    },
    transition: {
        default: '0.3s ease-in-out',
        fast: '0.1s ease-in-out',
    },
} as const
