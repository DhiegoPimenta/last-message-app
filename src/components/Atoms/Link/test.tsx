import { screen } from '@testing-library/react'
import theme from 'styles/theme'
import { renderWithTheme } from 'utils/tests/helpers'
import LinkTo from '.'

describe('<Link/>', () => {
    it('Should be render with default values', () => {
        const { container } = renderWithTheme(<LinkTo url="/">Link</LinkTo>)

        expect(screen.getByText(/link/i)).toBeInTheDocument()
        expect(screen.getByText(/link/i)).toHaveStyle({
            color: theme.colors.primary,
            'text-decoration': 'underline',
            cursor: 'pointer',
        })
        expect(container.firstChild).toMatchSnapshot()
    })

    it('Should be render with correctly url value', () => {
        renderWithTheme(<LinkTo url="/test">Link</LinkTo>)

        const link = screen.getByTestId('link-button')
        expect(link.getAttribute('href')).toBe('/test')
    })
})
