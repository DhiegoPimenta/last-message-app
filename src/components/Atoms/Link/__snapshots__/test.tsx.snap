// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link/> Should be render with default values 1`] = `
.c0 {
  color: #64CBC5;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-size: 2.4rem;
  cursor: pointer;
}

<a
  class="c0"
  data-testid="link-button"
  href="/"
>
  Link
</a>
`;
